File: dart/isogram/lib/isogram.dart
class Isogram {
  bool isIsogram(String word) {
    String filtered = word.toLowerCase().replaceAll(new RegExp(r"\W"), '');
    return filtered.split('').toSet().length == filtered.length;
  }

  bool myOriginalSolution(String word) {
    // when I made this, I had forgotten about sets.
    RegExp regExp = RegExp(r"(\w)");
    Map<String, int> charCounts = new Map();
    bool iso = true;

    for (String c in word.toLowerCase().split('')) {
      if (regExp.hasMatch(c)) {
        if (charCounts.containsKey(c)) {
          iso = false;
          break;
        } else {
          charCounts[c] = 1;
        }
      }
    }
    return iso;
  }
}

File: csharp/isogram/Isogram.cs
using System;
using System.Collections.Generic;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        Dictionary<char, int> charDict = new Dictionary<char, int>();

        foreach (var letter in word.ToLower())
        {
            if (letter != ' '  && letter != '-')
            {
                if (charDict.ContainsKey(letter))
                {
                    return false;
                }
                else
                {
                    charDict[letter] = 1;
                }
            }
            
        }

        return true;
    }
}
